examples/scripts/3d_equation_cylindric.py

```diff
@@ -0,0 +1,176 @@
+import matplotlib.pyplot as plt
+import matplotlib.tri as tri
+import numpy as np
+from scipy.interpolate import griddata
+
+import pybamm
+from pybamm import (
+    BaseModel,
+    Discretisation,
+    Scalar,
+    ScikitFiniteElement3D,
+)
+from pybamm.meshes.scikit_fem_submeshes_3d import ScikitFemGenerator3D
+
+R_outer = 0.6
+R_inner = 0.1
+H = 1.0
+alpha = 0.1
+
+model = BaseModel()
+
+r = pybamm.SpatialVariable("r", ["current collector"], coord_sys="cylindrical polar")
+theta = pybamm.SpatialVariable(
+    "theta", ["current collector"], coord_sys="cylindrical polar"
+)
+z = pybamm.SpatialVariable("z", ["current collector"], coord_sys="cylindrical polar")
+T = pybamm.Variable("T", domain="current collector")
+
+model.algebraic = {T: alpha * pybamm.laplacian(T)}
+model.initial_conditions = {T: Scalar(25)}
+model.boundary_conditions = {
+    T: {
+        "r_min": (Scalar(100), "Dirichlet"),
+        "r_max": (Scalar(0), "Dirichlet"),
+        "z_min": (Scalar(0), "Neumann"),
+        "z_max": (Scalar(0), "Neumann"),
+    }
+}
+
+geometry = {
+    "current collector": {
+        r: {"min": pybamm.Scalar(R_inner), "max": pybamm.Scalar(R_outer)},
+        z: {"min": pybamm.Scalar(0), "max": pybamm.Scalar(H)},
+    }
+}
+
+submesh_types = {
+    "current collector": ScikitFemGenerator3D(geom_type="cylinder", h=0.08)
+}
+
+var_pts = {r: None, theta: None, z: None}
```

Comment:
especially here, it would be good to explain more about how the number of elements is chosen in 3D meshes. can we visualise the mesh?